ext/src/nassau.rs

```diff
@@ -130,6 +135,67 @@ impl MilnorSubalgebra {
     pub fn iter_signatures(&self, degree: i32) -> impl Iterator<Item = Vec<PPartEntry>> + '_ {
         SignatureIterator::new(self, degree)
     }
+
+    fn top_degree(&self) -> i32 {
+        self.profile
+            .iter()
+            .map(|&entry| 2i32.pow(entry as u32) - 1)
+            .enumerate()
+            .map(|(idx, entry)| (2i32.pow(idx as u32 + 1) - 1) * entry)
+            .sum()
+    }
+
+    fn optimal_for(s: u32, t: i32) -> MilnorSubalgebra {
+        let mut subalgebras = vec![MilnorSubalgebra::zero_algebra()];
+        for subalgebra in SubalgebraIterator::new() {
+            let coeff = 2i32.pow(subalgebra.profile.len() as u32) - 1;
+            if t < coeff * (s as i32 + 1) + subalgebra.top_degree() {
+                // (s,t) is not in the vanishing region of `subalgebra` or any further subalgebra
+                break;
+            }
+            subalgebras.push(subalgebra);
+        }
+        subalgebras.pop().unwrap()
```

Comment:
I don't understand why you have a `subalgebras` vector if we only care about the last element. Wouldn't the following make more sense:
```suggestion
        let mut result = MilnorSubalgebra::zero_algebra();
        for subalgebra in SubalgebraIterator::new() {
            let coeff = 2i32.pow(subalgebra.profile.len() as u32) - 1;
            if t < coeff * (s as i32 + 1) + subalgebra.top_degree() {
                // (s,t) is not in the vanishing region of `subalgebra` or any further subalgebra
                break;
            }
            result = subalgebra;
        }
        result
```